test/testdata/dsl/t_struct_override.rb

```diff
@@ -0,0 +1,17 @@
+# typed: true
+
+class Override < T::Struct
+  extend T::Sig
+
+  prop :foo, String
+
+  sig {params(foo: Integer).void}
+  def initialize(foo:)
+    @foo = foo.to_s
+  end
+end
+
+Override.new # error: Missing required keyword argument `foo` for method `Override#initialize`
+Override.new(foo: "no") # error: Expected `Integer` but found `String("no")` for argument `foo`
+Override.new(foo: 3, bar: 4) # error: Unrecognized keyword argument `bar` passed for method `Override#initialize`
+T.reveal_type(Override.new(foo: 3).foo) # error: Revealed type: `String`
```

Comment:
Do you want to add a test that errors for incompatible override, and assert that the error happens on their `initialize` instead of the class def loc?

Reply:
Sure.